### PR TITLE
Resources adjustments, some url changes, probes fine-tuning, initContainers, declarativeness etc

### PR DESCRIPTION
**Pull request by baranbartu**
### TL;DR
Main idea is to prepare the charts for the next step with minimal change: more declarative approach (avoiding commenting/uncommenting things, letting user deploy the stack with minimum effort).

### Things that have been done in this PR
- Letting user to choose a RWO access moded storages (in case there is a single node k8s setup - convenient for default setup as well)
- minimum resource usages are adjusted after trying multiple deployment attempts
- backend's ALLOWED_HOSTS are adjusted for in-cluster access
- liveness/readinessProbs are adjusted for better startup
- frontend startup depends on backend's readiness (an initContainer setup is added for frontend)
- uncommenting/commentings things in values are no longer exists, instead used a more declarative approach

PS. Please feel free to provide feedback on this as the next steps are blocked until a decision is made on this.

### CODE REVIEW

File: helm/ray/templates/raycluster-storage.yaml [hunk 1]
```diff
@@ -5,8 +5,8 @@ metadata:
...
 spec:
   accessModes:
-  - ReadWriteMany
+  - {{ .Values.storage.accessMode | default "ReadWriteMany" }}
   resources:
     requests:
       storage: 20Gi
-  storageClassName: {{ .Values.storage.storageClassName | default "default" }}
\ No newline at end of file
+  storageClassName: {{ .Values.storage.storageClassName | default "default" }}
```
[the remark this answers is not in the document]
Comment (baranbartu) [reply helm/ray/templates/raycluster-storage.yaml:12]: Then if that's the case we will need to make this more declarative by configuring it through values.yaml, and as an entrypoint, via cli. We will need to ask that question to user whether they want to do it manually, then I'd say we don't put `storageClassName` if that's case, we need to ask user to provide `claim ref`  and we will need to add some lines like below in PVC to match th emanually craeted PV

```
  selector:
    matchLabels:
      pv-label-key: pv-label-value
```

this will match the manually created PV.

But I'd suggest us to move on with the current structure by not making significant changes, And once it is done with basics, we could move on with customs and adjust our setup accordingly.

WDYT?

File: helm/syntho-ui/templates/backend/backend-service.yaml [hunk 2]
```diff
@@ -13,7 +13,7 @@ metadata:
 spec:
   ports:
     - name: backend-port
-      port: {{ .Values.backend.service.port }}
+      port: 80
```
[the remark this answers is not in the document]
Comment (baranbartu) [reply helm/syntho-ui/templates/backend/backend-service.yaml:16]: Previous frontend setup (the broken one) was trying to reach backend via a real domain. I changed it to in-cluster domain with previous port like `svc.namespace.blahblah:backendcontainerport`  then frontend was brokebn, Then I saw this and made it 80 in order access directly via in-cluster dns without providing explicit port. 

I'd say since this is related to service's exposed IP's 80th port which is like an encapsulated network for only this specific service, it is convenient to leave it like this. But if you have hard opinion on that, I could revert it.

File: helm/ray/templates/raycluster-storage.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (baranbartu) [reply helm/ray/templates/raycluster-storage.yaml]: made some changes on PVC setup,could you pls take a look at it?

File: helm/syntho-ui/templates/backend/backend-service.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (Younday) [reply helm/syntho-ui/templates/backend/backend-service.yaml:16]: Yeah it's fine for now, I'll mark it as resolved!

File: helm/syntho-ui/templates/dbs/database-data-2-persistentvolumeclaim.yaml [hunk 3]
```diff
@@ -12,5 +12,6 @@ spec:
   resources:
     requests:
       storage: 100Mi
+  storageClassName: {{ .Values.db.storageClassName | default "default" }}
```
Comment (Younday) [inline helm/syntho-ui/templates/dbs/database-data-2-persistentvolumeclaim.yaml:15]: Here we have a default value of "default", but in the `values.yaml` we set it to `standard`. Should this be either the same default, or change the one in `values.yaml`?

**Review by Younday — COMMENTED** [review]
Just a couple of small comments/questions. Let me know whether something is unclear!

File: helm/ray/templates/raycluster-storage.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Younday) [reply helm/ray/templates/raycluster-storage.yaml]: That looks fine to me btw! I think that should cover that situation correctly

**Comment by Younday** [comment]
I'll start some testing on the changes on my end tomorrow as well, I just want to run it locally and one of our Kubernetes clusters as well before including it. It will also require some changes in our documentation, so I'll start on those as well.

File: helm/syntho-ui/templates/dbs/database-data-2-persistentvolumeclaim.yaml (hunk 3 above)
Comment (baranbartu) [reply helm/syntho-ui/templates/dbs/database-data-2-persistentvolumeclaim.yaml:15]: it was made `default` in values.yaml

**Review by Younday — APPROVED** [review]
